src/content_scripts/applyColorSchemeJs.js

```diff
@@ -22,109 +51,294 @@ const ADDON_FAKED_WARNING = "matchMedia has been faked/overwritten by add-on web
  */
 function getColorTypeFromMediaQuery(mediaQueryString) {
     // to avoid expensive RegEx, first use a simple check
-    if (!mediaQueryString.includes(MEDIA_QUERY_COLOR_SCHEME)) {
+    if (!hasColorMediaQuery(mediaQueryString)) {
         return null;
     }
 
     if (MEDIA_QUERY_PREFER_COLOR[COLOR_STATUS.LIGHT].test(mediaQueryString)) {
         return COLOR_STATUS.LIGHT;
     } else if (MEDIA_QUERY_PREFER_COLOR[COLOR_STATUS.DARK].test(mediaQueryString)) {
         return COLOR_STATUS.DARK;
-    } else if (MEDIA_QUERY_PREFER_COLOR[COLOR_STATUS.NO_PREFERENCE]) {
+    } else if (MEDIA_QUERY_PREFER_COLOR[COLOR_STATUS.NO_PREFERENCE].test(mediaQueryString)) {
         return COLOR_STATUS.NO_PREFERENCE;
     } else {
         return null;
     }
 }
 
 /**
- * Returns a fake MediaQueryList as best as possible.
+ * Whether a string contains media query for color scheme
  *
  * @private
- * @param {string} mediaQueryString the original media query string
- * @param {COLOR_STATUS} askedFor
- * @returns {Object} looks like MediaQueryList
+ * @param {string} mediaQueryString
+ * @returns {boolean}
  */
-function fakeMediaQueryResult(mediaQueryString, askedFor) {
-    let matches = false;
+function hasColorMediaQuery(mediaQueryString) {
+    return (mediaQueryString.includes(MEDIA_QUERY_COLOR_SCHEME));
+}
 
-    // only return true, if the asked status is the same as the one we want to fake
-    if (fakedColorStatus === askedFor) {
-        matches = true;
+/**
+ * Evaluate a media query string.
+ * Returns null if the query has nothing to do with color.
+ *
+ * @private
+ * @param {string} mediaQueryString
+ * @returns {boolean|null}
+ */
```

Comment:
This could be a place to keep a reference to the original MDN doc:

```suggestion
  * @see {@link https://developer.mozilla.org/docs/Web/API/Window/matchMedia}
  */
```

Reply:
Hard to say, I don't think that link properly describes that function.
I think a link to https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList at the top of the file is more appropriate.

Reply:
Okay, then feel free to do so.